src/uproot/writing/writable.py

```diff
@@ -36,14 +39,11 @@
 from uproot._util import no_filter, no_rename
 
 
-def create(file_path, **options):
+def create(file_path: str | IO, **options):
```

Comment:
I'm not sure if this annotations make sense. I think its confusing that the variable is called `file_path` but it also supports file-like objects (it took me some time to realise that I could just pass file-like objects).

I am in favour of renaming this variable to something like `file_path_or_object` or atleast anotate it to make it clear, but I'm not sure if this is the correct way to annotate it.

Reply:
Yes, this argument was historical—the ability to use objects instead of paths was added later. A name like `file_path_or_object` is rather long; I'd be tempted to use a word like `source` except that it's the name of a class (so I'd then expect the `source` to be a `Source`). Perhaps `src` would be a good choice.

_Except_ that this argument is passed around a lot of functions. If it has one name (`file_path`) in some functions and another (`src`) in others, that would increase the confusion, rather than decrease it. Finding out what is the correct set of function arguments to update would be a project in itself.

Adding a type hint makes short work of that, just as the documentation did (though the type hint is more visible than the documentation).